src/Sarif/RuleUtilities.cs

```diff
@@ -32,7 +33,7 @@ public static Result BuildResult(FailureLevel level, IAnalysisContext context, R
             return BuildResult(level, kind, context, region, ruleMessageId, arguments);
         }
 
-        public static Result BuildResult(FailureLevel level, ResultKind kind, IAnalysisContext context, Region region, string ruleMessageId, params string[] arguments)
+        public static Result BuildResult(FailureLevel level, ResultKind kind, IAnalysisContext context, Region region, string ruleMessageId, params object[] arguments)
```

Comment:
>object [](start = 148, length = 6)

Not sure I like this change. it's preferable to require callers to convert to strings, so that they can apply culture insensitive/sensitive transformations during ToString. we can't make an appropriate choice from the object.ToString() helper.